### ISSUE TITLE
PoX Updates: `stack-extend` and `delegate-stack-extend`

### ISSUE THREAD

**Issue by kantai** [stacks-network/stacks-blockchain#2532]
As part of updating PoX in Stacks 2.1, the PoX contract should implement:

```
;; extend the "tx-sender"s active lockup
(define-public (stack-extend (extend-count uint) (start-at uint)))
;; extend the delegated active lockup
(define-public (delegate-stack-extend (stacker principal) (extend-count uint) (start-at uint)))
```

in order to support extensions of existing Stacking locks.

**Comment by hstove** [comment]
- Is `start-at` a cycle or a block height? 
- Why not only have the parameter `extend-count`, which extends from the end of the last currently stacked cycle?
- Can `stack-extend` be called at any time during a current stacking position? i.e. if I'm locked for 12 cycles, and in the 4th cycle, can I `stack-extend`?
  - Is it safe to assume that I'll never be able to `stack-extend` that would result in my last cycle being more than 12 blocks in the future? i.e. if I have 8 cycles left, I can only call `(stack-extend u4)`?

**Comment by jcnelson** [comment]
Implemented with #2755
